### ISSUE TITLE
[Feature Request] Write results in the note

### ISSUE THREAD

**Issue by RobertoHRO** [gtg922r/obsidian-numerals#22]
First of all - I have been waiting for so long for such a plugin! - Thank you so much!
Here is my feature request: 
Can you write the results of the calculations including the unit in the note? The feature could look like this:
![image](https://user-images.githubusercontent.com/25647024/210585364-4831e1f4-aeec-4d84-b15a-aadbe6680d2f.png)
All the results the user wants to be displayed will be written to the note:
![image](https://user-images.githubusercontent.com/25647024/210585212-f5fe7e2b-9465-49e3-b0c2-c871189f851b.png)

If an existing note with math-blocks is opened and the math-blocks are evaluated by _obsidian-numerals_ and the results in the note and from  _obsidian-numerals_ differ - you could indicate this and ask the user for permission to update.

**Comment by gtg922r** [comment]
I've been thinking about this a ton, I definitely agree this feature needs to exist. It bothers me the results aren't saved into the plain text. 

I need to think about the right syntax for when to do this. As you indicate I think using the result annotation syntax might be a good  place to start.

Adding autocomplete is almost done and I think I might take this on next.

**Comment by akehrer** [comment]
I also wanted to say thank your for this plugin.

Using the result annotation would be similar to how [Calca](http://calca.io) handles it, so the consistency there would be nice to have. Another plus is that now all the information is there when copy/pasting or sharing the raw markdown file.

**Comment by mofosyne** [comment]
By the way, I'm a bit iffy about using `=>` for results annotation (looks too similar to C equal or more than operator). 

Just wondering what you think about `:=` as result annotator indicator instead? Plus it would give the user an option between showing just the results on the fly or making sure the results is recorded into the text file as well.

That's how I've been manually recording results from https://speqmath.com/ which puts their results annotation in a new line and I wanted to keep it more compact in my personal notes (before this plugin existed).

e.g.

```
    a = 1+1 := 2
    b = $3
    c = a * b := $6 AUD
```

-----

Also you may want to add some consistent attribute metadata when using results annotation so you know when the last time it was run (and any other enviromental variables that may be lost if that block is copied and pasted). This would be useful when opening the file in a non obsidian text editor.

The adhoc standard is shown in https://talk.commonmark.org/t/consistent-attribute-syntax/272

I've double checked if adding `{}` would interfere with obsidian rendering of codeblocks and it did not break anything still.

    ```math {lastupdated="2000-01-23T01:23:45.678+09:00"}
    a = 1+1 := 2
    b = $3 := $3 AUD
    c = a * b := $6 AUD
    ```

**Comment by RobertoHRO** [comment]
To check if I have understood you correctly, is this what you intend to do?  
`c = a * b := $6` writes to the mardown file and renders the result in the obsidian reader/life view. And `c = a * b => $6` only renders the result in the obsidian reader/life view.

--- 

I like the suggestion of putting the last modified timestamp in the file. It is also helpful (may be necessary) when trying to automatically modify the md file.

**Comment by mofosyne** [comment]
Yes that's correct on both count (I do wonder if this plugin has boolean operators and if so what would it be...)

**Comment by rightdroid** [comment]
> By the way, I'm a bit iffy about using `=>` for results annotation (looks too similar to C equal or more than operator).
> 
> Just wondering what you think about `:=` as result annotator indicator instead? Plus it would give the user an option between showing just the results on the fly or making sure the results is recorded into the text file as well.

Wanted to throw my 2 cents in here regarding this suggestion. `:=` is used as a variable assignment operator for some languages (AHK, Golang), while the _greater/less than or equal_ is always the other way around. For alternatives, perhaps `->` or `>>`, but current one is sensible as well, IMHO.

Btw, superb plugin, @gtg922r 💯

**Comment by gtg922r** [comment]
Thanks for the thoughts on syntax all. I settled in a syntax that will expand to other numerals features, and thats using `@[]`. Unfortunately, I admit the syntax is not particularly intuitive, and I'm open to considering other options. 

I've added experimental support in [Numerals Beta 1.4](https://github.com/gtg922r/obsidian-numerals/releases/tag/1.4.0). Wrapping any variable name in `@[]`, for example `@[meaningOfLife]`, will insert the value of the variable after `::` when the block next renders. The raw text of the note will be modified to say: `@[meaningOfLife::42]`. This has the intentional side-effect of allowing you to set dataview values on a page using numerals. 

Please see the release notes and I'd love your help testing out this feature! Please see the README for instructions how to beta test using BRAT
